packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -482,7 +482,16 @@ class InputDecorator extends StatelessWidget {
       );
     }
 
-    Widget inputChild;
+    Widget inputChild = new KeyedSubtree(
+      // It's important that we maintain the state of our child subtree, as it
+      // may be stateful (e.g. containing text selections). Since our build
+      // function risks changing the depth of the tree, we preserve the subtree
+      // using global keys.
+      // GlobalObjectKey(context) will always be the same whenever we are built.
+      key: new GlobalObjectKey(context),
```

Comment:
Used a subclass of GlobalObjectKey to salt it, and added some documentation to GlobalObjectKey itself documenting the trick of using a private subclass to salt.